src/main/scala/subsystem/BaseSubsystem.scala

```diff
@@ -134,7 +134,21 @@ abstract class BaseSubsystem(implicit p: Parameters) extends BareSubsystem
     }
   }
 
-  lazy val logicalTreeNode = new SubsystemLogicalTreeNode()
+  lazy val logicalTreeNode = {
+    val subsystemLogicalTreeNode = new SubsystemLogicalTreeNode()
+    val builtInDevices = Seq(
+      pbus.builtInDevices,
+      fbus.builtInDevices,
+      mbus.builtInDevices,
+      cbus.builtInDevices
+    )
+    for (builtIn <- builtInDevices) {
+      builtIn.errorOpt.foreach { error =>
+        LogicalModuleTree.add(subsystemLogicalTreeNode, error.logicalTreeNode)
+      }
+    }
```

Comment:
Not sure if this is more readable, but how about:

```suggestion
    for (builtIn <- builtInDevices; error <- builtIn.errorOpt) {
      LogicalModuleTree.add(subsystemLogicalTreeNode, error.logicalTreeNode)
    }
```
?

Reply:
looking at this again, I'm not sure why I didn't just do this to be consistent
```suggestion
    builtInDevices.foreach { builtIn =>
      builtIn.errorOpt.foreach { error =>
        LogicalModuleTree.add(subsystemLogicalTreeNode, error.logicalTreeNode)
      }
    }
```
but I dunno. I guess the your version looks the best to me.

Reply:
I think @albertchen-sifive 's is more what I'm used to in the code base, but either change looks good and symmetrical :)

Reply:
more importantly. I just realized that we are overriding this field. so I guess this stuff should go in an `attachBuiltInDevicesLogicalTreeNodes` method that can be called in the constructor of the LogicalTreeNode?

Reply:
Ah, I had forgotten about the `lazy val` being overridden by subclasses. Now that you've brought it up, I don't actually think we want any of this stuff happening in the constructor of the LogicalTreeNode. Is there a problem with just moving all of the `LogicalModuleTree.add()` statements below/outside of the `lazy val logicalTreeNode = { }` statement?

Reply:
that would force evaluation of the `lazy val logicalTreeNode` early? doesn't seem like it matters when the logicalTreeNode gets evaluated though, because it still works if I move it outside of the block.

Reply:
Yeah, that should be fine, since we have a pattern of doing this in a number of places. The logicalTreeNodes are only really meant to create the tree of nodes without fulling evaluated the nodes, somewhat analogous to Diplomatic nodes.

Reply:
I don't get how `lazy val`s work